cloud/smartdatacenter/sdc.py

```diff
@@ -0,0 +1,318 @@
+    required: false
+    default: "yes"
+    choices: [ "yes", "no" ]
+    aliases: []
+  wait_timeout:
+    description:
+      - how long before wait gives up, in seconds
+    default: 600
+    aliases: []
+  state:
+    description:
+      - create or delete instances
+    required: false
+    default: 'present'
+    aliases: []
+
+requirements: [ "smartdc" ]
+author: Preston Marshall
+'''
+
+EXAMPLES = '''
+
+# Provision single machine example
+- local_action:
+    module: sdc
+    account: myaccount
+    key_id: /myaccount/keys/mykey
+    name: my-virtual-machine
+    package: 486bb054-6a97-4ba3-97b7-2413d5f8e849
+    image: 62f148f8-6e84-11e4-82c5-efca60348b9f
+    location: 'us-east-1.api.joyentcloud.com'
+
+# Ensure 5 machines with tag "webserver" are running, does not destroy existing machines
+- local_action:
+    module: sdc
+    account: myaccount
+    key_id: /myaccount/keys/mykey
+    package: 486bb054-6a97-4ba3-97b7-2413d5f8e849
+    image: 62f148f8-6e84-11e4-82c5-efca60348b9f
+    location: 'us-east-1.api.joyentcloud.com'
+    count_tag:
+      role: webserver
+    exact_count: 5
+
+# Delete machines by their tags
+- local_action:
+    module: sdc
+    account: myaccount
+    key_id: /myaccount/keys/mykey
+    location: 'us-east-1.api.joyentcloud.com'
+    tags:
+      role: webserver
+    state: absent
+
+# Delete machine by its id
+- local_action:
+    module: sdc
+    account: myaccount
+    key_id: /myaccount/keys/mykey
+    machine_id: 51307f2f-02a3-4bdc-9b38-2f263eb43e2d
+    location: 'us-east-1.api.joyentcloud.com'
+    state: absent
+'''
+
+import os
+import sys
+import time
+try:
+    import smartdc
+    from smartdc import DataCenter
+    HAS_SMARTDC = True
+except ImportError:
+    HAS_SMARTDC = False
+
+def _wait_for_status(module, machines, status):
+    wait_timeout = int(module.params.get('wait_timeout'))
+    wait_timeout_time = time.time() + wait_timeout
+    while not all( sm.status() == status for sm in machines ) and wait_timeout_time > time.time():
+        time.sleep(2)
+    # Check again that all are in expected state. If not, we timed out
+    if not all( sm.status() == status for sm in machines ):
+        module.fail_json(msg="Timed out creating machines")
+
+def list_existing_machines(module, sdc, tags):
+    if not tags:
+        module.fail_json(msg="Must have tags to list existing machines for deletion/counting")
+    machines = sdc.machines(tags=tags, state='running')
+    return machines
+
+
+def create_virtual_machine(module, sdc):
+    """
+    Create new virtual machine
+
+    module : AnsibleModule object
+    sdc: authenticated sdc Datacenter object
+
+    Returns:
+        True if a new machine was created, false otherwise
+    """
+    name = module.params.get('name')
+    package = module.params.get('package')
+    image = module.params.get('image')
+    networks = module.params.get('networks')
+    wait = module.params.get('wait')
+    wait_timeout = int(module.params.get('wait_timeout'))
```

Comment:
no need to cast if you already requested a type in module definition (as is below)